### ISSUE TITLE
Color output

### ISSUE THREAD

**Issue by rylev** [fermyon/spin#1481]
Currently spin's CLI output is almost exclusively in the default text color of the terminal it is in. This can lead to large walls of mono-colored output which is hard to parse. Most modern CLI tools output in color (when the functionality is available). I believe spin should as well. 

## Considerations 

There are certain considerations we should keep in mind:

* We should use color sparingly as if we use it too much, we end up right back in the situation we currently find ourselves in: hard to parse output.
* We should use color consistently. For example, if `Error:` is colored red in some output, it should be colored red in all output.
* We should not produce color escape codes when there is not terminal available, when the user has certain environment variables set (e.g., `NO_COLOR`), etc.

## Next steps

I believe it would be very difficult to design a grand vision for color and just go execute on it. We should think about how we want to apply color incrementally while still following the considerations laid out above. 

To that end I think there are some ideas we can start with. After implementing these ideas, we can evaluate how the user experience has improved (or not) and determine if we want to continue, revert, or stop making changes to the color output. 

Possible ideas include:
* Coloring `Error` in error messages red
* Coloring the first word of steps in `spin up` and `spin build` green
* Coloring warnings yellow (or at least adding a yellow "Warning" to them).

**Comment by itowlson** [comment]
Accessibility should also be a consideration.  Naively, I'd guess that colouring at least won't make things worse that our current monochrome output.  But 1. we should validate that it doesn't and 2. if we can find options that play nicely with screen readers / colour blindness / etc. then that should be a consideration.

cc @flynnduism for graphic design and information layout sensibilities

**Comment by rylev** [comment]
Going to close this as complete since we have color for various things now. We can open smaller issues for any further improvements.
